tower-spawn-ready/src/service.rs

```diff
@@ -26,34 +25,20 @@ enum Inner<T> {
     Future(oneshot::Receiver<Result<T, Error>>),
 }
 
-impl<T> SpawnReady<T, DefaultExecutor> {
+impl<T> SpawnReady<T> {
     /// Creates a new `SpawnReady` wrapping `service`.
-    ///
-    /// The default Tokio executor is used to drive service readiness, which
-    /// means that this method must be called while on the Tokio runtime.
     pub fn new(service: T) -> Self {
-        Self::with_executor(service, DefaultExecutor::current())
-    }
-}
-
-impl<T, E> SpawnReady<T, E> {
-    /// Creates a new `SpawnReady` wrapping `service`.
-    ///
-    /// `executor` is used to spawn a new `BackgroundReady` task that is
-    /// dedicated to driving the inner service to readiness.
-    pub fn with_executor(service: T, executor: E) -> Self {
         Self {
-            executor,
             inner: Inner::Service(Some(service)),
         }
     }
 }
 
-impl<T, E, Request> Service<Request> for SpawnReady<T, E>
+impl<T, Request> Service<Request> for SpawnReady<T>
 where
     T: Service<Request> + Send,
     T::Error: Into<Error>,
-    E: BackgroundReadyExecutor<T, Request>,
+    DefaultExecutor: BackgroundReadyExecutor<T, Request>,
```

Comment:
don't think you need this?

Reply:
You do, otherwise the call to `spawn` doesn't work (or you'd have to `Box::pin` the future).